RadarSDK/remodel/RadarBeacon.m

```diff
@@ -0,0 +1,152 @@
+
+NS_ASSUME_NONNULL_BEGIN
+
+@implementation RadarBeacon
+
+- (nullable instancetype)initWithCoder:(NSCoder *)aDecoder {
+    if ((self = [super init])) {
+        __id = (id)[aDecoder decodeObjectForKey:k_idKey];
+        __description = (id)[aDecoder decodeObjectForKey:k_descriptionKey];
+        _numberMetadata = (id)[aDecoder decodeObjectForKey:kNumberMetadataKey];
+        _metadataArray = (id)[aDecoder decodeObjectForKey:kMetadataArrayKey];
+        _location = (id)[aDecoder decodeObjectForKey:kLocationKey];
+        _version = (id)[aDecoder decodeObjectForKey:kVersionKey];
+    }
+    return self;
+}
+
+- (nullable instancetype)initWithRadarJSONObject:(nullable id)object {
+    if (!object || ![object isKindOfClass:[NSDictionary class]]) {
+        return nil;
+    }
+    NSDictionary *dictionary = (NSDictionary *)object;
+    if ((self = [super init])) {
+        if (!dictionary[@"_id"]) {
+            return nil;
+        }
+        __id = (NSString *)dictionary[@"_id"];
+        __description = (NSString *)dictionary[@"_description"];
+        if (!dictionary[@"numberMetadata"]) {
+            return nil;
+        }
+        _numberMetadata = (NSDictionary *)dictionary[@"numberMetadata"];
+        if (!dictionary[@"metadataArray"]) {
+            return nil;
+        }
+        _metadataArray = (NSArray *)dictionary[@"metadataArray"];
+        if (!dictionary[@"location"]) {
+            return nil;
+        }
+        _location = [[RadarLocation alloc] initWithRadarJSONObject:dictionary[@"location"]];
+        if (!dictionary[@"version"]) {
+            return nil;
+        }
+        _version = (NSNumber *)dictionary[@"version"];
+    }
+    return self;
+}
+
+- (instancetype)initWith_id:(NSString *)_id
+               _description:(nullable NSString *)_description
+             numberMetadata:(NSDictionary<NSString *, NSNumber *> *)numberMetadata
+              metadataArray:(NSArray<NSString *> *)metadataArray
+                   location:(RadarLocation *)location
+                    version:(NSNumber *)version {
+    RMParameterAssert(_id != nil);
+    RMParameterAssert(numberMetadata != nil);
+    RMParameterAssert(metadataArray != nil);
+    RMParameterAssert(location != nil);
+    RMParameterAssert(version != nil);
+    if ((self = [super init])) {
+        __id = [_id copy];
+        __description = [_description copy];
+        _numberMetadata = [numberMetadata copy];
+        _metadataArray = [metadataArray copy];
+        _location = [location copy];
+        _version = [version copy];
+    }
+
+    return self;
+}
+
+- (id)copyWithZone:(nullable NSZone *)zone {
+    return self;
+}
+
+- (NSDictionary *)dictionaryValue {
+    NSMutableDictionary *dict = [[NSMutableDictionary alloc] init];
+    dict[@"_id"] = __id;
+    if (__description) {
+        dict[@"_description"] = __description;
+    }
+    dict[@"numberMetadata"] = _numberMetadata;
+    dict[@"metadataArray"] = _metadataArray;
```

Comment:
we could even support the case when we have a collection of nested model objects, if we add category method to make NSArray/NSDictionary conforming to `RadarJSONCoding` protocol